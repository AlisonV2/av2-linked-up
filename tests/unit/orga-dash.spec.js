import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import OrgaDashboard from '@/components/organizers/OrgaDashboard';
import AppTitle from '@/components/app/Title';
import AppButton from '@/components/app/Button';


const actions = {
  getOrgaProfile: jest.fn(),
  setOrgaThumbnail: jest.fn(),
  setOrgaProfile: jest.fn(),
};

const state = {
  Orgaprofile: {
    OrgaProfile: {
      name: 'Alison',
      city: 'Bordeaux',
      zip: '33800',
      thumbnail: 'url',
    },
  },
};

const store = createStore({
  state,
  actions,
});

const $router = {
  push: jest.fn(),
};

const $route = {
  params: { id: '1' },
};

const config = {
  data() {
    return {
      previewImage: null,
      profile: {
        name: 'Alison',
        description: 'Unicorn power',
        city: 'Bordeaux',
        zip: '33800',
        tattooed: 'Yes',
        thumbnail: 'url',
      },
      showSuccessToast: false,
      showErrorToast: false,
      fileSelected: null,
      fileErr: null,
    };
  },
  global: {
    plugins: [store],
    components: {
      'app-title': AppTitle,
      'app-button': AppButton
    },
    mocks: {
      $router,
      $route,
    },
  },
};
/**
 * @module OrgaDashboardTest
 */
describe('OrgaDashboard.vue', () => {
  it('OrgaDashboard Snapshot', () => {
    const component = shallowMount(OrgaDashboard, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if getOrgaProfile action is called', async () => {
    const component = shallowMount(OrgaDashboard, config);
    OrgaDashboard.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getOrgaProfile).toHaveBeenCalled;
  });

  it('Check if setOrgaProfile action is called', async () => {
    const component = shallowMount(OrgaDashboard, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect(actions.setOrgaProfile).toHaveBeenCalled;
  });

  it('Check if setOrgaThumbnail action is called', async () => {
    const component = shallowMount(OrgaDashboard, config);
    component.find('.thumbnail-button').trigger('click');
    await component.vm.$nextTick();
    expect(actions.setOrgaThumbnail).toHaveBeenCalled;
  });

  it('setPreviewImage should be called', () => {
    const component = shallowMount(OrgaDashboard, config);
    const file = component.find('input[type="file"]');
    file.trigger('change');
    expect(component.vm.setPreviewImage).toHaveBeenCalled;
  });
});
