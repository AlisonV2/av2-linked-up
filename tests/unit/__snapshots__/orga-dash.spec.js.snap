// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgaDashboard.vue OrgaDashboard Snapshot 1`] = `
<div
  class="row"
>
  <app-title-stub />
  <!--v-if-->
  <!--v-if-->
  <div
    class="col-12 col-lg-4 mt-4"
  >
    <div
      class="mt-3"
    >
      <input
        type="file"
      />
      <div
        class="error"
      />
    </div>
    <img
      alt="Profile Image"
      class="img-fluid"
      height="250"
      src=""
      width="250"
    />
    <div
      class="btn-center mt-3 thumbnail-button"
    >
      <app-button-stub
        mode="save-btn"
      />
    </div>
    <div
      class="error"
    />
  </div>
  <form
    class="col-12 col-lg-8 mt-4"
  >
    <div
      class="form-floating mb-3"
    >
      <input
        class="form-control"
        placeholder="Public Name"
        type="text"
      />
      <label>
        Public Name
      </label>
    </div>
    <div
      class="input-group"
    >
      <div
        class="form-floating mb-3 col-lg-6"
      >
        <input
          class="form-control"
          placeholder="City"
          type="text"
        />
        <label>
          City
        </label>
      </div>
      <div
        class="form-floating mb-3 col-lg-6"
      >
        <input
          class="form-control"
          placeholder="Zip Code"
          type="text"
        />
        <label>
          Zip Code
        </label>
      </div>
    </div>
    <div
      class="col-12 btn-right mt-3"
    >
      <div
        class="btn-group mb-3"
      >
        <app-button-stub />
      </div>
    </div>
  </form>
</div>
`;
